packages/components/utils/src/useKeyboard/useKeyboard.ts

```diff
@@ -0,0 +1,55 @@
+import { useState, useEffect, useCallback } from 'react';
+
+export interface UseKeyboardProps {
+  /**
+   * @description defines which element is the event attached to
+   * @default document
+   */
+  ref?: Document | Window | HTMLElement;
+  /**
+   * @description defines which key to look for i.e. `ArrowUp`, `Escape`, `Shift`
+   */
+  key: KeyboardEvent['key'];
+  /**
+   * @description defines the attached event type
+   * @default 'keyup'
```

Comment:
Why `keyup` is a default? 
IMO `keydown` would be/should be more common.

Reply:
I recall reading something about a11y and that `keyup` allow users to cancel events with `esc` but I can't find any resources. I can validate that and change if my assumption is incorrect 👍🏼